localstack/services/awslambda/invocation/runtime_environment.py

```diff
@@ -0,0 +1,188 @@
+import logging
+import random
+from datetime import date, datetime
+from enum import Enum, auto
+from threading import RLock, Timer
+from typing import TYPE_CHECKING, Dict, Literal, Optional
+
+from localstack import config
+from localstack.services.awslambda.invocation.executor_endpoint import ServiceEndpoint
+from localstack.services.awslambda.invocation.lambda_models import FunctionVersion
+from localstack.services.awslambda.invocation.runtime_executor import RuntimeExecutor
+from localstack.utils.strings import to_str
+
+if TYPE_CHECKING:
+    from localstack.services.awslambda.invocation.version_manager import QueuedInvocation
+
+STARTUP_TIMEOUT_SEC = 10.0
+HEX_CHARS = [str(num) for num in range(10)] + ["a", "b", "c", "d", "e", "f"]
+
+LOG = logging.getLogger(__name__)
+
+
+class RuntimeStatus(Enum):
+    INACTIVE = auto()
+    STARTING = auto()
+    READY = auto()
+    RUNNING = auto()
+    FAILED = auto()
+    STOPPED = auto()
+
+
+InitializationType = Literal["on-demand", "provisioned-concurrency"]
+
+
+class InvalidStatusException(Exception):
+    def __init__(self, message: str):
+        super().__init__(message)
+
+
+def generate_runtime_id() -> str:
+    return "".join([random.choice(HEX_CHARS) for _ in range(32)])
+
+
+class RuntimeEnvironment:
+    runtime_executor: RuntimeExecutor
+    status_lock: RLock
+    status: RuntimeStatus
+    initialization_type: InitializationType
+    last_returned: datetime
+    startup_timer: Optional[Timer]
+
+    def __init__(
+        self,
+        function_version: FunctionVersion,
+        initialization_type: InitializationType,
+        service_endpoint: ServiceEndpoint,
+    ):
+        self.id = generate_runtime_id()
+        self.status = RuntimeStatus.INACTIVE
+        self.status_lock = RLock()
+        self.function_version = function_version
+        self.initialization_type = initialization_type
+        LOG.debug("Creating new executor")
```

Comment:
maybe add the runtime ID to the debug statements?

Reply:
a those were not only debug statements, but debugging statements. No real value in them, will remove